bpf/process/bpf_process_event.h

```diff
@@ -11,7 +11,8 @@
 
 #define ENAMETOOLONG 36 /* File name too long */
 
-#define MAX_BUF_LEN 256
+#define MATCH_BINARIES_PATH_MAX_LENGTH 256
```

Comment:
Do these lengths present a problem in terms of potential truncation?

Reply:
`MATCH_BINARIES_PATH_MAX` is the size of the paths users can specify in `values` under `MatchBinaries`. That is,  it only limits the size of the user-specified paths that we want to match against.

For example,

```
      - matchBinaries:
        - operator: "In"
          values:
          - "/usr/bin/bash"
 ```
 
 would work, but:
 
```
      - matchBinaries:
        - operator: "In"
          values:
          - "'/foo/foo/foo/foo/foo/foo/foo/foo/foo/foo/foo/foo/foo/foo/foo/foo/foo/foo/foo/foo/foo/foo/foo/foo/foo/foo/foo/foo/foo/foo/foo/foo/foo/foo/foo/foo/foo/foo/foo/foo/foo/foo/foo/foo/foo/foo/foo/foo/foo/foo/foo/foo/foo/foo/foo/foo/foo/foo/foo/foo/foo/foo/foo/foo"
 ```
 
 wouldn't.

Reply:
Thank you for this work @kkourt! This will help us out immensely. 

One comment, could we in the future perhaps convert this to use the same string handling as was added in #2069 ? Not for now, just in the future.

Reply:
Yap, I've also thought the same :)

Reply:
so this is an issue on the prefix feature because of the way we read dentries, since we read them from end to start, you can escape a prefix match using a long path (>256). There are attempts to fix at the moment (https://github.com/cilium/tetragon/pull/2718) but the only durable fix would be to extend `MATCH_BINARIES_PATH_MAX_LENGTH` to 4096 somehow.

Reply:
If we ifdef'd into 6.x kernels maybe 4096 is fine with better BPF? This way limitation only exists on <X.Y versions?

Reply:
So I've been trying to extend the length to 4096, mostly making prepend_name (the function we use to read dentry) to support 4096 chars and I'm currently blocked when the compiler is replacing `memset` and `memcpy` when the object (from a map value) is too large.